src/modules/ChannelList/dux/reducers.ts

```diff
@@ -108,22 +109,57 @@ export default function channelListReducer(
       .with({ type: channelListActions.ON_USER_LEFT }, (action) => {
         const { channel, isMe } = action.payload;
         const { allChannels, currentUserId, currentChannel, channelListQuery, disableAutoSelect } = state;
-        let nextChannels = allChannels.filter((ch) => ch.url !== channel.url);
-        let nextChannel = null;
-        if (channelListQuery) {
-          if (filterChannelListParams(channelListQuery, channel, currentUserId)) {
-            // Good to [add to/keep in] the ChannelList
+        let nextChannels = [...allChannels];
+        let nextChannel: GroupChannel = channel;
+
+        /**
+         * 1. If I left channel:
+         *   - Remove the channel from channel list
+         *   - Replace currentChannel with the next ordered channel
+         * 2. If other member left channel:
+         *   2-1. If query is given:
+         *     2-1-1. If channel no longer matches the query
+         *       - Same as step 1
+         *     2-1-2. If channel matches the query:
+         *       - Upsert channel list with the channel
+         *       - Replace currentChannel IFF url is same
+         *   2-2. If query is not given,
+         *     - Same as step 2-1-2
+         */
+        if (isMe) {
+          const channelAt = allChannels.findIndex((ch: GroupChannel) => ch.url === channel.url);
+          if (channelAt > -1) {
+            nextChannels.splice(channelAt, 1);
+            nextChannel = getNextChannel({
+              channel,
+              currentChannel,
+              allChannels,
+              disableAutoSelect,
+            });
+          }
+        } else if (channelListQuery) {
+          if (!filterChannelListParams(channelListQuery, channel, currentUserId)) {
+            const channelAt = allChannels.findIndex((ch: GroupChannel) => ch.url === channel.url);
+            if (channelAt > -1) {
+              nextChannels.splice(channelAt, 1);
+              nextChannel = getNextChannel({
+                channel,
+                currentChannel,
+                allChannels,
+                disableAutoSelect,
+              });
+            }
+          } else {
             nextChannels = getChannelsWithUpsertedChannel(allChannels, channel, state.channelListQuery?.order);
+            if (currentChannel?.url === channel.url) {
+              nextChannel = channel;
+            }
+          }
+        } else {
+          nextChannels = getChannelsWithUpsertedChannel(allChannels, channel, state.channelListQuery?.order);
+          if (currentChannel?.url === channel.url) {
+            nextChannel = channel;
           }
```

Comment:
How about reducing lines like this?
```suggestion
if (isMe || (channelListFilter && !filterChannelListParams(channelListQuery, channel, currentUserId))) {
const channelAt = allChannels.findIndex((ch: GroupChannel) => ch.url === channel.url);
          if (channelAt > -1) {
            nextChannels.splice(channelAt, 1);
            nextChannel = getNextChannel({
              channel,
              currentChannel,
              allChannels,
              disableAutoSelect,
            });
          }
} else {
nextChannels = getChannelsWithUpsertedChannel(allChannels, channel, state.channelListQuery?.order);
          if (currentChannel?.url === channel.url) {
            nextChannel = channel;
          }
}
```